src/api/api_core.php

```diff
@@ -111,6 +104,26 @@ function auth_session_exists() {
     return FALSE;
 }
 
+function update_session($userId, $userName) {
+    session_regenerate_id(TRUE);
+    $_SESSION['user_id'] = $userId;
+    $_SESSION['user_name'] = $userName;
+    $_SESSION['user_lastactive'] = time();
+}
+
+function set_authorisation_cookies($userId, $key, $doStayLoggedIn) {
+    if ($doStayLoggedIn) {
+        // an expiry time of one year in the future should be sufficient
+        $expiryTime = time() + 365 * 24 * 60 * 60;
+    } else {
+        $expiryTime = 0;
+    }
+
+    setcookie('auth_userid', $userId, $expiryTime, '/', '', FALSE);
+    setcookie('auth_key', $key, $expiryTime, '/', '', FALSE);
+    setcookie('expiry_time', $expiryTime, $expiryTime, '/', '', FALSE);
```

Comment:
IMO it would be less confusing if we named this cookie something like `stay_logged_in`, to make it clear what it's for, and that we're the ones controlling it.  As is, i had some confusion while reading the code, thinking maybe `expiry_time` was a magical cookie name, because of the reference to the `$expiryTime` variable on other cookies.